Core/Python/update_firmware_using_dup.py

```diff
@@ -106,19 +106,27 @@ def get_group_list(ip_address, headers):
 
 
 def get_device_list(ip_address, headers):
-    """ Get list of devices from OME """
-    device_list = None
-    device_url = 'https://%s/api/DeviceService/Devices' % (ip_address)
-    response = requests.get(device_url, headers=headers, verify=False)
-    if response.status_code == 200:
-        dev_response = response.json()
-        if dev_response['@odata.count'] > 0:
-            device_list = [x['Id'] for x in dev_response['value']]
-        else:
-            print ("No devices found at ", ip_address)
-    else:
-        print ("No devices found at ", ip_address)
-    return device_list
+	""" Get list of devices from OME """
+	ome_device_list = None
+	next_link_url = 'https://%s/api/DeviceService/Devices' % ip_address
+	while next_link_url is not None:
+		device_response = requests.get(next_link_url, headers=headers, verify=False)
+		next_link_url = None
+		if device_response.status_code == 200:
+			dev_json_response = device_response.json()
+			if dev_json_response['@odata.count'] <= 0:
+				print("No devices found at ", ip_address)
+				return
+
+			if '@odata.nextLink' in dev_json_response:
+				next_link_url = 'https://%s/' %ip_address + dev_json_response['@odata.nextLink']
+
+			if dev_json_response['@odata.count'] > 0:
+				ome_device_list = [x['Id'] for x in dev_json_response['value']]
```

Comment:
This might collect the devices only from the last call. Think we should collect the data from all responses in the loop and then finally collect all the ids.